polaris/benchmark/_base.py

```diff
@@ -35,6 +35,9 @@
     TargetType,
 )
 
+# Type variable for the return type of evaluate
+BenchmarkResultsType = TypeVar("BenchmarkResultsType")
```

Comment:
Nice addition with this! I'd suggest:

```suggestion
BenchmarkResultsType = TypeVar("BenchmarkResultsType", BenchmarkResultsV1, BenchmarkResultsV2)
```

So we can get some extra type safety and bind `BenchmarkResultsType` to either `BenchmarkResultsV1` or `BenchmarkResultsV2`.